src/typechecker.erl

```diff
@@ -129,8 +137,10 @@ any_subtype([Ty1|Tys], Ty, TEnv) ->
 %% The function compat_ty is just a convenience function to be able to
 %% pattern match on types in a nice way.
 compat(T1, T2, A, TEnv) ->
-    Ty1 = typelib:remove_pos(normalize(T1, TEnv)),
-    Ty2 = typelib:remove_pos(normalize(T2, TEnv)),
+    Ty1 = normalize(T1, TEnv),
+    Ty2 = normalize(T2, TEnv),
+    ?assert_normalized_anno(Ty1),
+    ?assert_normalized_anno(Ty2),
```

Comment:
I was expecting the assertions to be checked before normalization. Isn't the invariant that you're trying to establish that all types have their annotations normalized **before** calling `normalize`?

Reply:
OK. I was thinking that the only place where the annotations need to be zeroed is for the call to `sets:is_element({Ty1, Ty2}, A)`, but `remove_pos` was done before `normalize` before, so I'll keep that order. I don't think normalize/2 adds any non-zero annotations.

The assertion just check the top-level anno though, so it's not bullet proof. Just something for the transition.